### PR TITLE
Add test environment without any extras installed

### PR DESCRIPTION
**Pull request by sk1p**
Updates to ensure everything outside the `udf` module can actually be used without installing the full LiberTEM dependency.

# TODO

- [x] refactor code not to import `libertem` directly
   - [x] in `base`
   - [x] in `common`
   - [x] in the tests (other than those in `tests/udf/`)
- [x] verify the fixture changes that make `libertem` optional

## Contributor Checklist:

* [x] I have added myself to [the creators.json file](https://github.com/LiberTEM/LiberTEM-blobfinder/blob/master/packaging/creators.json)
* [x] I have added [a changelog entry](https://github.com/LiberTEM/LiberTEM-blobfinder/tree/master/docs/source/changelog) for my contribution
* [x] I have added/updated documentation for all user-facing changes
* [x] I have added/updated test cases

### CODE REVIEW

**Comment by sk1p** [comment]
@uellue @matbryan52 this is the CI addition mentioned in #61 - let me know your thoughts on this, and also feel free to take over the PR

**Comment by matbryan52** [comment]
I removed the `common` extras group as it no longer makes sense after these changes. If needed it could be retained as an empty group for backwards compatibility?

**Comment by sk1p** [comment]
> I removed the `common` extras group as it no longer makes sense after these changes. If needed it could be retained as an empty group for backwards compatibility?

I don't really think it is important to be kept, as long as we mention it in the changelog (and update the install docs)

File: conftest.py [hunk 1]
```diff
@@ -80,17 +75,26 @@ def auto_libs(doctest_namespace):
...
+        import libertem.masks
+        import libertem.api
+        import libertem_blobfinder
+
+        doctest_namespace["libertem"] = libertem
+        doctest_namespace["libertem.utils"] = libertem.utils
+        doctest_namespace["libertem.utils.generate"] = libertem.utils.generate
+        doctest_namespace["libertem.masks"] = libertem.masks
+        doctest_namespace["libertem.api"] = libertem.api
+        doctest_namespace["libertem_blobfinder"] = libertem_blobfinder
```
Comment (sk1p) [inline conftest.py:98]: Should these also use the `libertem_blobfinder` locations? Are these even still used in doctests?

If not:

```suggestion
        import libertem
        import libertem.api
        import libertem_blobfinder

        doctest_namespace["libertem"] = libertem
        doctest_namespace["libertem.api"] = libertem.api
        doctest_namespace["libertem_blobfinder"] = libertem_blobfinder
```

**Review by sk1p — COMMENTED** [review]
@matbryan52 thank you for taking over!

Should the examples also be updated to replace their usage of `libertem.masks` / `libertem.utils`, to make them consistent with the rest of the code? Added a small comment inline, other than that, it looks good to me!

**Comment by sk1p** [comment]
@matbryan52 thanks for finishing this, I'll go ahead and merge!

**Pull request merged**